packages/api-server/src/filter-web3-tx.ts

```diff
@@ -0,0 +1,359 @@
+import { HexString } from "@ckb-lumos/base";
+import { Script } from "@ckb-lumos/base";
+import { Hash, HexNumber } from "@ckb-lumos/base";
+import {
+  HexU32,
+  U32,
+  L2Transaction,
+  L2TransactionReceipt,
+  GodwokenClient,
+  schemas,
+  U64,
+} from "@godwoken-web3/godwoken";
+import { Reader } from "ckb-js-toolkit";
+import { envConfig } from "./base/env-config";
+import { EthTransaction, EthTransactionReceipt } from "./base/types/api";
+import { Uint128, Uint32, Uint64 } from "./base/types/uint";
+import { PolyjuiceSystemLog, PolyjuiceUserLog } from "./base/types/gw-log";
+
+const CKB_SUDT_ACCOUNT_ID: U32 = 1;
+
+const EMPTY_ADDRESS = "0x" + "00".repeat(20);
+
+const GW_LOG_POLYJUICE_SYSTEM = "0x2";
+const GW_LOG_POLYJUICE_USER = "0x3";
+
+const PENDING_TRANSACTION_INDEX = "0x0";
+
+const DEFAULT_LOGS_BLOOM = "0x" + "00".repeat(256);
+
+export async function filterWeb3Transaction(
+  txHash: Hash,
+  rpc: GodwokenClient,
+  tipBlockNumber: U64,
+  tipBlockHash: Hash,
+  l2Tx: L2Transaction,
+  l2TxReceipt?: L2TransactionReceipt
+): Promise<[EthTransaction, EthTransactionReceipt | undefined] | undefined> {
+  const pendingBlockHash = bumpHash(tipBlockHash);
+  const pendingBlockNumber = new Uint64(tipBlockNumber + 1n).toHex();
+
+  const fromId: U32 = +l2Tx.raw.from_id;
+  const fromScriptHash: Hash | undefined = await rpc.getScriptHash(fromId);
+  if (fromScriptHash == null) {
+    return undefined;
+  }
+  const fromScript: Script | undefined = await rpc.getScript(fromScriptHash);
+  if (fromScript == null) {
+    return undefined;
+  }
+
+  if (fromScript.code_hash !== envConfig.ethAccountLockHash) {
```

Comment:
will it be ok if we don't add this check? right now we have tron lock, and the web3-indexer include tx both for eth lock and tron lock (to support calling a eth-tx using tron wallet for signing)

Reply:
or add check for tron lock too ?
```suggestion
  if (fromScript.code_hash !== envConfig.ethAccountLockHash && fromScript.code_hash !== envConfig.tronAccountLockHash) {
```

Reply:
updated.